qemu/tests/libvirt_host_model_test.py

```diff
@@ -0,0 +1,157 @@
+    name = uuid.uuid4().hex
+    try:
+        path.find_command("virsh")
+    except path.CmdNotFoundError:
+        test.cancel("Virsh executable not set or found on path")
+
+    xml = """
+            <domain type='kvm'>
+                <name>%s</name>
+                <memory>1</memory>
+                <os>
+                    <type arch='%s' machine='%s'>hvm</type>
+                </os>
+                <cpu mode='host-model'/>
+            </domain>
+          """ % (name, vm_arch, machine)
+    xml_file = os.path.join(data_dir.get_tmp_dir(), "temp_xml_for_cpu")
+    with open(xml_file, "w") as f:
+        f.write(xml)
+    try:
+        logging.info("Get cpu model and features from virsh")
+        virsh.define(xml_file)
+        virsh.start(name)
+    except Exception as err:
+        test.cancel(err)
+    else:
+        qemu_cpu_info = cpu.get_cpu_info_from_virsh_qemu_cli(name)
+        libvirt_cpu_info = get_cpu_info_from_dumpxml(name)
+        params["qemu_cpu_info"] = qemu_cpu_info
+        params["libvirt_cpu_info"] = libvirt_cpu_info
+
+        cpu_model_qemu = qemu_cpu_info["model"]
+        cpu_model_libvirt = libvirt_cpu_info["model"]
+        if cpu_model_qemu != cpu_model_libvirt:
+            logging.error("mismatch cpu model bwteen qemu %s and libvirt %s",
+                          cpu_model_qemu, cpu_model_libvirt)
+            return False
+        params["cpu_model"] = cpu_model_qemu
+        qemu_cpu_flags = cpu.parse_qemu_cpu_flags(qemu_cpu_info['flags'])
+        libvirt_cpu_flags = libvirt_cpu_info['features']
+        qemu_cpu_flags = extend_flags_patterns(qemu_cpu_flags)
+        miss_flags = []
+        mismatch_flags = []
+        for flag in libvirt_cpu_flags.keys():
+            if flag not in qemu_cpu_flags.keys():
+                if libvirt_cpu_flags[flag] == 'on':
+                    miss_flags.append(flag)
+            elif libvirt_cpu_flags[flag] != qemu_cpu_flags[flag]:
+                mismatch_flags.append(flag)
+        if miss_flags or mismatch_flags:
+            logging.error("\nmiss flags %s from qemu cli\n", miss_flags)
+            logging.error("\nmismatch flags %s between libvirt and qemu\n",
+                          mismatch_flags)
+            return False
+        return True
+    finally:
+        if virsh.is_alive(name):
+            virsh.destroy(name, ignore_status=True)
+        virsh.undefine(name, ignore_status=True)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu models with full flags that libvirt generates
+    1) start a default configuration xml by virsh
+    2) get the cpu elements from libvirt xml by 'virsh dumpxml $domain'
+    3) parse the cpu_flags and store into libvirt_cpu_flags
+    4) get the cpu flags from qemu process and stroe to qemu_cpu_flags
+    5) Compare compare the cpu flags, model of xml and qemu process, should be same
+    6) Boot guest with qemu_cpu_flags, model, guest works well
```

Comment:
@yanan-fu, Hi, would please help to check Yumei's comment? thanks.

Reply:
@luckyh  Call for your review for this issue.

Reply:
> Hi @yanan-fu , I prefer @nanliu-r 's, since it will add more flags defined globally, e.g. hyper-v flags. That's more likely the scenario used by uplayer. Thanks.

IMO, this opinion is infer the cause from the conclusion, It is not sufficient from my view.
My understanding is:  If this test case just call the same functions in the framework, compare and get qemu cpu info, i am ok with any, repeat operations are acceptable to me.
However, it is not.  This case define the xml template by itself and use it for compare.  (evenif the template is same in tp-qmeu and avocado-vt).

Reply:
> > Hi @yanan-fu , I prefer @nanliu-r 's, since it will add more flags defined globally, e.g. hyper-v flags. That's more likely the scenario used by uplayer. Thanks.
> 
> IMO, this opinion is infer the cause from the conclusion, It is not sufficient from my view. My understanding is: If this test case just call the same functions in the framework, compare and get qemu cpu info, i am ok with any, repeat operations are acceptable to me. However, it is not. This case define the xml template by itself and use it for compare. (evenif the template is same in tp-qmeu and avocado-vt).

Will take Yanan's advice and update patch, thanks.